snuba/snuba_migrations/uptime_monitor_checks/0001_uptime_monitor_checks.py

```diff
@@ -0,0 +1,96 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import UUID, Column, DateTime, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import AddIndicesData, OperationTarget, SqlOperation
+
+storage_set = StorageSetKey.UPTIME_MONITOR_CHECKS
+table_prefix = "uptime_monitor_checks"
+local_table_name = f"{table_prefix}_local"
+dist_table_name = f"{table_prefix}_dist"
+
+## what about all the fancy codecs? do we need those?
+columns: List[Column[Modifiers]] = [
+    Column("organization_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("environment", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("uptime_subscription_id", UInt(64)),
+    Column("uptime_check_id", UUID()),
+    Column("scheduled_check_time", DateTime()),
+    Column("timestamp", DateTime()),
+    Column("_sort_timestamp", DateTime()),
+    Column("duration", UInt(64)),
+    Column("region_id", UInt(16, Modifiers(nullable=True))),
+    Column("check_status", String(Modifiers(low_cardinality=True))),
+    Column(
+        "check_status_reason",
+        String(Modifiers(nullable=True, low_cardinality=True)),
+    ),
+    Column("http_status_code", UInt(16)),
+    Column("trace_id", UUID()),
+    Column("retention_days", UInt(16)),
+]
+
+indices: Sequence[AddIndicesData] = [
+    AddIndicesData(
+        name="bf_trace_id",
+        expression="trace_id",
+        type="bloom_filter",
+        granularity=1,
+    )
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=storage_set,
+                table_name=local_table_name,
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    # do i actually need primary key to be different than sorting key?
+                    primary_key="(organization_id, project_id, uptime_subscription_id, _sort_timestamp, uptime_check_id, trace_id)",
```

Comment:
do we actually need trace_id here?

Reply:
The EAP RPC has a way to query by `trace_id`. If you're not putting a `trace_id` somewhere, the requests on this endpoint will be slower.

Also, you don't need to set the sort key and the primary key. What we usually want is the primary key to be as small as possible (it's in RAM) and there's no need to be overly specific since, at some point, there will be so little data compared to what needs to be scanned.

In your case, I would put the timestamp right after `project_id` because, after this, how many `uptime_subscription_id` will you have? if it's something like under 8k, it might fit into one part and scanning the primary key then the part is not faster than just the part.

Reply:
>The EAP RPC has a way to query by trace_id. If you're not putting a trace_id somewhere, the requests on this endpoint will be slower.

i've added a bloom filter index in the migration.

>Also, you don't need to set the sort key and the primary key. What we usually want is the primary key to be as small as possible (it's in RAM) and there's no need to be overly specific since, at some point, there will be so little data compared to what needs to be scanned.

thank you. modified.